test/jdk/javax/swing/JComboBox/bug4180054.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 1998, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Robot;
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.event.ListDataEvent;
+import javax.swing.event.ListDataListener;
+
+/*
+ * @test
+ * @bug 4180054
+ * @summary Tests that DefaultComboBoxModel doesn't fire a "contents changed" unneccesarily
+ * @key headful
+ * @run main bug4180054
+ */
+
+public class bug4180054 {
+    static JFrame frame;
+    static JComboBox comboBox;
+    static int numberOfContentsChangedEvents = 0;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(() -> createTestUI());
+        robot.waitForIdle();
+        robot.delay(250);
+
+        // change selected index 3 times
+        SwingUtilities.invokeAndWait(() -> {
+            comboBox.setSelectedIndex(1);
+            comboBox.setSelectedIndex(3);
+            comboBox.setSelectedIndex(2);
```

Comment:
guess old test also checks "If the same item is selected twice in a row, the number should not increase"
so I guess we should also include another `comboBox.setSelectedIndex(2`) and `numberOfContentsChangedEvents ` should still be 3